File: sdk/php/src/events/RemoveFragments.php
<?php
/**
 * @copyright Copyright (c) PutYourLightsOn
 */

namespace starfederation\datastar\events;

use starfederation\datastar\Consts;
use starfederation\datastar\enums\EventType;

class RemoveFragments implements EventInterface
{
    use EventTrait;

    public string $selector;
    public int $settleDuration = Consts::DEFAULT_FRAGMENTS_SETTLE_DURATION;
    public bool $useViewTransition = Consts::DEFAULT_FRAGMENTS_USE_VIEW_TRANSITIONS;

    public function __construct(string $selector, array $options = [])
    {
        $this->selector = $selector;

        foreach ($options as $key => $value) {
            $this->$key = $value;
        }
    }

    /**
     * @inerhitdoc
     */
    public function getEventType(): EventType
    {
        return EventType::RemoveFragments;
    }

    /**
     * @inerhitdoc
     */
    public function getDataLines(): array
    {
        $dataLines = [
            $this->getDataLine(Consts::SELECTOR_DATALINE_LITERAL, $this->selector),
        ];

        if ($this->settleDuration !== Consts::DEFAULT_FRAGMENTS_SETTLE_DURATION) {
            $dataLines[] = $this->getDataLine(Consts::SETTLE_DURATION_DATALINE_LITERAL, $this->settleDuration);
        }

        if ($this->useViewTransition !== Consts::DEFAULT_FRAGMENTS_USE_VIEW_TRANSITIONS) {
            $dataLines[] = $this->getDataLine(Consts::USE_VIEW_TRANSITION_DATALINE_LITERAL, $this->getBooleanAsString($this->useViewTransition));
        }

        return $dataLines;
    }
}

File: sdk/java/src/main/java/StarFederation/Datastar/events/RemoveFragments.java
package StarFederation.Datastar.events;

import StarFederation.Datastar.enums.EventType;
import StarFederation.Datastar.events.AbstractSSEEvent;
import static StarFederation.Datastar.Consts.*;

import java.util.ArrayList;
import java.util.List;

public class RemoveFragments extends AbstractSSEEvent {

    private final String selector;
    private final RemoveFragmentsOptions removeFragmentsOptions;

    public RemoveFragments(String selector, RemoveFragmentsOptions removeFragmentsOptions) {
        super();
        if (selector == null || selector.isBlank()) {
            throw new IllegalArgumentException("Selector cannot be null or empty.");
        }
        this.selector = selector.trim();
        this.removeFragmentsOptions = removeFragmentsOptions != null
                ? removeFragmentsOptions
                : RemoveFragmentsOptions.create();
    }

    @Override
    public EventType getEventType() {
        return EventType.RemoveFragments;
    }

    public String[] getDataLines() {
        List<String> dataLines = new ArrayList<>();

        dataLines.add(getDataLine(SELECTOR_DATALINE_LITERAL, selector));

        addDataLineIfNotDefault(dataLines, SETTLE_DURATION_DATALINE_LITERAL,
                removeFragmentsOptions.getSettleDuration(), DEFAULT_FRAGMENTS_SETTLE_DURATION);

        if (removeFragmentsOptions.isUseViewTransition()) {
            dataLines.add(getDataLine(USE_VIEW_TRANSITION_DATALINE_LITERAL, getBooleanAsString(true)));
        }

        return dataLines.toArray(String[]::new);
    }
}
